evm/src/main/java/org/hyperledger/besu/evm/precompile/ECRECPrecompiledContract.java

```diff
@@ -97,13 +128,22 @@ public PrecompileContractResult computePrecompile(
       final Optional<SECPPublicKey> recovered =
           signatureAlgorithm.recoverPublicKeyFromSignature(h, signature);
       if (recovered.isEmpty()) {
-        return PrecompileContractResult.success(Bytes.EMPTY);
+        res =
+            new PrecompileInputResultTuple(
+                nonMutatedInput, PrecompileContractResult.success(Bytes.EMPTY));
+        ecrecCache.put(nonMutatedInput.hashCode(), res);
```

Comment:
We can calculate the hashcode only once above, as done in other precompiles.

Reply:
AbstractBytes.hashCode() caches but MutableBytes.hashCode() does not.  Several of the inputs I looked at were actually MutableBytes, so I made all of the precompiles pre-calculate and hang onto the cacheKey as you suggested 👍